src/daemon/dbus_service.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2019 Luxoft Sweden AB
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#include "daemon/dbus_service.h"
+
+#include <giomm.h>
+#include <glibmm.h>
+
+#include "common/dbus.h"
+
+namespace TemplateDBusService::Daemon
+{
+    DBusService::DBusService(const Glib::RefPtr<Glib::MainLoop> &main_loop) : main_loop_(main_loop)
+    {
+    }
+
+    DBusService::~DBusService()
+    {
+        unown_name();
+    }
+
+    void DBusService::own_name()
+    {
+        if (connection_id_ != 0)
+            return;
+
+        connection_id_ = Gio::DBus::own_name(Gio::DBus::BUS_TYPE_SYSTEM,
+                                             Common::DBus::TEMPLATE_SERVICE_NAME,
+                                             sigc::mem_fun(*this, &DBusService::bus_acquired),
+                                             sigc::mem_fun(*this, &DBusService::name_acquired),
+                                             sigc::mem_fun(*this, &DBusService::name_lost));
+    }
+
+    void DBusService::unown_name()
```

Comment:
Ha ha, fun naming: 'unown' :). A bit difficult to read/understand, consider rename method to not_own_name() ?

Reply:
Hum, I think it should be kept. It makes sense if you look at the GDBus API. E.g.:

Reply:
I think I prefer `release_name()` over `not_own_name()`.

Reply:
Perhaps it's bad to use another terminology. I was reading the code too fast. I thought unown was a new term but apparently it wasn't. So perhaps you should keep own/unown, sorry Martin :(.

Reply:
Reverted commits that changed name. Resolving.